CONTRIBUTING.rst

```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
    guidelines <https://github.com/deepchecks/deepchecks/blob/master/CONTRIBUTING.rst>`__
 -  Check if your changes are consistent with the
    `guidelines <#general-guidelines-for-contribution>`__
--  Changes are consistent with the `coding
-   style <#coding-style>`__
--  Run the `unit
-   tests <#running-unit-tests>`__
+-  Refer to the `Windows users <#windows-users>`__ section if your development environment
+   is Windows.
+
+   - `Creating Virtual Environment <#creating-virtual-environment>`__
+   - `Installing Dependencies <#installing-dependencies>`__
+   - `Running Unit Tests <#windows-running-unit-tests>`__
+   - `Coding Style <#windows-coding-style>`__
+
+-  Refer to the `Linux/Mac users <#linux-and-mac-users>`__ section if your development environment
+   is either Linux or Mac.
+
+   - `Running Unit Tests <#linux-mac-running-unit-tests>`__
+   - `Coding Style <#linux-mac-coding-style>`__
+   - `Test Coverage <#linux-mac-test-coverage>`__
+   - `Generate Docs <#linux-mac-generate-docs>`__
 
 General guidelines for contribution
 ------------------------------------
 
--  Include unit tests when you contribute new features, as they help to
-   a) prove that your code works correctly, and b) guard against future
-   breaking changes to lower the maintenance cost.
+-  Include unit tests when you contribute new features, as they help to:
+   
+   - Prove that your code works correctly.
+   - Guard against future breaking changes to lower the maintenance cost.
+
 -  Bug fixes also generally require unit tests, because the presence of
    bugs usually indicates insufficient test coverage.
+-  Make sure that your code changes complies with the coding style of the
+   project.
 -  Keep API compatibility in mind when you change code. Reviewers of
    your pull request will comment on any API compatibility issues.
 
+Windows Users
+---------------
+For contributors using Windows operating system, you have to manually run
+the following commands since as of now the ``make`` commands works with
+Linux/Mac OS. **BTW, you are open to solve this issue!!**
+
+
+----------------------------
+Creating virtual environment
```

Comment:
Please use the title standard that exists in other rst files. Usually the double ----- titles are "higher" than the single ----- titles.

Reply:
Done